### ISSUE TITLE
Premessa-output files-FlowJo naming problems

### ISSUE THREAD

**Issue by mleipold** [ParkerICI/premessa#4]
I finally got output daughter files from the debarcoder.  Note:  these had already gone through the Premessa Normalizer.

However, when I dragged them into FlowJo, there were problems:

1.  They all had the same Name.  If I added a column for "File Name" (one of the FJ defaults), the appropriate daughter debarcode names *did* appear.  Therefore, they *are* being written to the file, but not in the place that FJ expects.

2.  Some parameter naming is now a bit screwy.  Anything with only one "name" in the Parent file is now doubled.
Parent:  Time
Daughter:  Time :: Time

Same thing with Event_length, Center, Offset, Residual, Width, Residual, and beadDist.

Going back and comparing the Parent to Normalized, this appears to happen at the level of the Normalization (ie, before Debarcoding).  Normalized and Debarcoded are consistent, as far as I can tell.

![daughter-fj parameters](https://user-images.githubusercontent.com/1111114/28443243-b8620980-6d69-11e7-8cc1-6a9c0f37c7ab.png)
![parent-fj parameters](https://user-images.githubusercontent.com/1111114/28443244-b868ba28-6d69-11e7-9e6b-5c81772b562e.png)
![premessa-debarcoder-fcs in fj name issue](https://user-images.githubusercontent.com/1111114/28443249-c8a60134-6d69-11e7-87d3-417f670af0ee.png)

**Comment by pfgherardini** [comment]
The parameter name thing you mention is on purpose. When the $PxS parameter is missing, the software will copy the $PxN value in the $PxS. I think this is the best approach because there are a number of downstream analysis programs that have problems when the $PxS is missing (citrus was one of them for instance, although this has been fixed now). Unless this is creating serious problems, I'd rather keep it that way.

With regards to the file names. You mean that the actual FCS files have different names on disk, but appear all with the same name when loaded in FlowJo? If that's the case I suspect it may be due to the fact that I am not updating one of the FCS keywords and FlowJo is looking in that keyword to determine the file name, instead of taking the file name on disk. Can you put up on a shared folder a couple of those FCS files so that I can double check?

**Comment by rfinck** [comment]
This is probably due to the $FIL keyword not being updated. You can check it in R with the following:

```
library(flowCore)
fcs <- read.FCS('/path/to/your/file')
description(fcs)['$FIL']
```

**Comment by pfgherardini** [comment]
Should be fixed in the latest commit. @mleipold can you please reinstall the software and test it?

**Comment by mleipold** [comment]
![072417-norm only-name vs filename-still incorrect](https://user-images.githubusercontent.com/1111114/28544367-b808ce22-7077-11e7-8451-5a80d7d14c95.png)

Hi Federico,

It does not seem to be fixed.

I took the same two original *.FCS files (directly off Helios; note uppercase), and did the Premessa Normalization on them.  This gave me two *_normalized.fcs files (note lowercase).

I then put the two originals and the two daughters into Mac FJ10.1.r5.  

The "Name" is still coming up as *.FCS (uppercase) with no "_normalized".  If I add a column and look at "File Name", then "File Name" lists the correct file naming (pairs of *.FCS and *_normalized.fcs)

In other words, looking only at "Name" (the FJ default), there's no way to tell original from daughter.


I have attached a screenshot, including showing that Finder is listing the daughter files correctly as "*_normalized.fcs".

**Comment by pfgherardini** [comment]
Hi Mike,

Sorry about that. I don't have access to a copy of FlowJo so testing that is not easy for me.

It should be fixed in the latest version. Can you please install it again and test? By the way to reinstall the package you only need these lines

```
library(devtools)
install_github("ParkerICI/premessa")
```

**Comment by mleipold** [comment]
Hi Federico, @pfgherardini 

Here's a Drive folder with some files:  
https://drive.google.com/open?id=0Bz9sd8nC272jdHVhNnFaQ0dFTE0

For the comparison within FJ, I only looked at the "2x" file (original vs norm'd).

To confirm:  you want me to install it a second time today (I already reinstalled it once today, from your request last week)?

**Comment by pfgherardini** [comment]
Yes I was asking if you could reinstall again today and try to normalize the files again. I submitted a fix ~20 minutes ago that *should* fix the problem. Let me know.... if it still doesn't work I'll try and get my hands on a copy of FlowJo

**Comment by mleipold** [comment]
@pfgherardini 

OK, did the reinstall a second time, just using the
library(devtools)
install_github("ParkerICI/premessa")

part as you stated.


Unfortunately, it's still not working.....same Name/File Name issue.

---------
I doubt this has anything to do with it, but I'm running R 3.3.1, and RStudio Version 1.0.153.  I notice during the RStudio execution, it states that 'shiny' and 'devtools' were written under R 3.3.2.

Since I can get the program to execute, I doubt that's a problem.  But, in the interest of troubleshooting, I wanted to mention it just in case.

**Comment by pfgherardini** [comment]
Ok let me look into this more. Thanks for mentioning the version differences but I don't think those represent the problem.

**Comment by rfinck** [comment]
It looks like the old file name is getting saved somewhere in the FCS file,
so searching the header/description for what keyword it's under it is might
be a helpful clue.

**Comment by pfgherardini** [comment]
That's what I did but it didn't seem to fix the problem. At least on my machine, the old file name is nowhere to be found after it goes through the normalizer

**Comment by rfinck** [comment]
Weird -- and of course you did that, sorry for repeating myself.

**Comment by pfgherardini** [comment]
Alright should be fixed for real now! :)

@mleipold  can you please install again and retry? Apologize for the back and forth....

**Comment by mleipold** [comment]
![072517-norm only-name vs filename-fixed](https://user-images.githubusercontent.com/1111114/28580601-d0b0958e-7114-11e7-91c7-c627a55961de.png)

Hooray, it appears to be fixed!  Thanks!

We've been discussing it only at the Normalizer level; does this propagate to the Debarcoder as well, or would that be a separate fix, if necessary?

**Comment by pfgherardini** [comment]
The fix will work across the board, so for the normalizer, debarcoder, and also FCS panel editor

**Comment by mleipold** [comment]
Great.  If I find some issue with it down the road, I'll add it to this thread.

But otherwise, I consider this closed.
